Rework `SwapTileGrab` to work across different outputs

nferhat opened the pull request:
Title. This pull request implements this by moving the responsibility of managing the grab from the `Workspace` to the `Space` so that we have easy access to the swapped tile at all times.

There are still some small issues with this approach notably:
- If there's part of a tile that's shown between two outputs, and release it on another, it will awkwardly disappear from the other outputs (except the one you dropped it at)
- Sometimes tiles might not pick up new output presentation framerates IE. not update in sync with the new output

nferhat commented:
@nnyyxxxx For now this works only with the master layout, but this should be much better

nnyyxxxx commented:
some bugs i noticed:

picking up a floating window makes it continuously smaller
cursor state doesn't reset once you let go on another output
if held between 2 outputs and then let go, the window will no longer show on the 2 outputs (while floating of course)

nferhat commented:
> picking up a floating window makes it continuously smaller
Fixed.

> cursor state doesn't reset once you let go on another output
Fixed too.

> if held between 2 outputs and then let go, the window will no longer show on the 2 outputs (only on 1) (while floating of course)
Limitation of how the workspace system works since there's no shared floating space between monitors. We do display it while the window is grabbed to show where you are moving it, but when you release it will only display on the output with the pointer.

nferhat commented:
Ugh, to get correct rendering I have to rewrite and re-organize `Tile::render` and `Tile::render_inner`, how annoying

nferhat commented:
https://github.com/user-attachments/assets/59e50953-08c5-4ead-86f0-3facdfec4d23

Still have to implement for `centered-master` layout, but this looks nice

nferhat commented:
Hmm, I am currently not a fan of how fullscreening is handling right now. A possible idea to implement is to replace the currently fullscreened window with the one you dragged and drop onto that workspace.

Will see if this works out.

nferhat commented:
Okay, its now finished and is working. @nnyyxxxx @adamperkowski if possible you could tell me if everything is right with this so I can merge it.

nnyyxxxx commented:
> Okay, its now finished and is working. @nnyyxxxx @adamperkowski if possible you could tell me if everything is right with this so I can merge it.

looks good to me

nferhat commented:
@nnyyxxxx finished adding centering the tile around the cursor
